sprocs/grading_jobs_insert.sql

```diff
@@ -34,17 +36,27 @@ BEGIN
     IF NOT FOUND THEN RAISE EXCEPTION 'no such submission_id: %', submission_id; END IF;
 
     -- ######################################################################
-    -- delegate the call
-
-    IF grading_method = 'Internal' THEN
-        grading_job := grading_jobs_insert_internal(submission_id, authn_user_id,
+    -- build up all grading jobs
+    IF grading_method_internal = False AND grading_method_external = False AND grading_method_manual = False THEN
+        RAISE EXCEPTION 'all grading methods set to false: (internal %s, external %s, manual %s)', grading_method_internal, grading_method_external, grading_method_manual;
+    END IF;
+        
+    -- delegate internal grading job ()
+    IF grading_method_internal = True THEN
+        RETURN NEXT grading_jobs_insert_internal(submission_id, authn_user_id,
                             new_gradable, new_broken, new_format_errors, new_partial_scores,
                             new_score, new_v2_score, new_feedback, new_submitted_answer,
                             new_params, new_true_answer);
-    ELSIF grading_method = 'External' OR grading_method = 'Manual' THEN
-        grading_job := grading_jobs_insert_external_manual(submission_id, authn_user_id);
-    ELSE
-        RAISE EXCEPTION 'unknown grading_method: %', grading_method;
     END IF;
+    
+    -- delegate external/manual grading job
+    IF grading_method_external = True OR grading_method_manual = True THEN
+        RETURN NEXT grading_jobs_insert_external_manual(submission_id, authn_user_id, 'External');
+    END IF;
```

Comment:
We had a discussion about running manual grading as a "grading job" that should be completed by a person rather than a machine. This code is incorrect since the type of grading job inserted should be 'external'/'manual' rather than always external.

However, I don't know if this was the final representation we used for manual grading internally (i.e. if there's a separate table for manual jobs).